src/core/i18n/translate.js

```diff
@@ -49,6 +44,12 @@ export default function translate(options = {}) {
 
     Translate.displayName = `Translate[${getDisplayName(WrappedComponent)}]`;
 
-    return Translate;
+    function mapStateToProps({ api, i18n }) {
+      return {
+        i18n: makeI18n(i18n, api.lang),
```

Comment:
Oh, are you suggesting I move the `makeI18n()` call into the HOC instead of leaving it in `mapStateToProps`?